components/coprocessor_plugin_api/src/plugin_api.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use super::storage_api::*;
+use std::fmt;
+
+/// Raw bytes of the request payload from the client to the coprocessor.
+pub type RawRequest = [u8];
+/// The response from the coprocessor encoded as raw bytes that are sent back to the client.
+pub type RawResponse = Vec<u8>;
+
+/// Error returned by a plugin.
+#[derive(Debug)]
+pub enum PluginError {
+    StorageError(StorageError),
+    /// Field for custom errors that appeared while executing a coprocessor request.
+    ///
+    /// Errors that are reported to the client via `PluginError::Other` should be of a type were
+    /// the client can't handle them. Thus, only the error's message is sent to the client.
+    /// If the error can be handled by the client, coprocessor plugins should perform proper error
+    /// reporting via their `data` fields.
+    /// TODO: Should we even include "non-recoverable user errors"?
+    Other(String),
```

Comment:
I added a dedicated field for unknown user errors that originate in a coprocessor plugin, but I'm not sure if we need it or if error handling should exclusively be done via `data` (encoded response from plugin).